src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,71 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { currentTodoSlice, isSelected } from '../../features/currentTodo';
+import { useAppDispatch, useAppSelector } from '../../app/store';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
 
 export const TodoModal: React.FC = () => {
+  const todoModal = useAppSelector(isSelected);
+  const dispatch = useAppDispatch();
+
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    if (todoModal) {
+      setIsLoading(true);
+      getUser(todoModal.userId)
+        .then(resp => setUser(resp))
+        .finally(() => setIsLoading(false));
+    }
+  }, [todoModal]);
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {isLoading ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              Todo #{todoModal?.id}
+            </div>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => {
+                dispatch(currentTodoSlice.actions.deleteCurrentTodo());
```

Comment:
Verify that `currentTodoSlice` contains the `deleteCurrentTodo` action. If this action is missing, it will cause errors when dispatching actions.